lib/web/app/transport.go

```diff
@@ -296,7 +296,7 @@ func dialAppServer(ctx context.Context, proxyClient reversetunnel.Tunnel, cluste
 		To:                    &utils.NetAddr{AddrNetwork: "tcp", Addr: reversetunnel.LocalNode},
 		OriginalClientDstAddr: originalDst,
 		ServerID:              fmt.Sprintf("%v.%v", server.GetHostID(), clusterName),
-		ConnType:              types.AppTunnel,
+		ConnType:              server.GetTunnelType(),
```

Comment:
The origin label isn't checked at upsert time, what can go wrong if a regular app is heartbeated with an `okta` origin?

Reply:
Adapt the original [PR](https://github.com/gravitational/teleport/pull/23688) to `UpsertAppServer`

Reply:
The original PR has been updated: https://github.com/gravitational/teleport/pull/23688